### PR TITLE
order multi-alter statements

### PR DESCRIPTION
**Pull request by JCOR11599**
When running multi-alter statements, MySQL reorders the alters. 
This PR adds a sort function to organize these statements; the precedence order is currently:
1. `RENAME COLUMN`
2. `DROP COLUMN`
3. `ADD COLUMN`
4. `ALTER INDEX`

Note: When this is supposed to work, it does, but when it doesn't our error messages are different than the ones returned by MySQL. There are existing alter tests that also return the incorrect error, but I don't think this is worth fixing right now.

I also unskipped one BrokenScriptTest and fix the expected results for the others.

Companion PR: https://github.com/dolthub/dolt/pull/5831

### CODE REVIEW

File: sql/parse/parse.go [hunk 1]
```diff
@@ -1293,6 +1294,31 @@ func convertMultiAlterDDL(ctx *sql.Context, query string, c *sqlparser.MultiAlte
 			return nil, err
 		}
 	}
+
+	// certain alter statements need to happen before others
+	sort.Slice(statements, func(i, j int) bool {
```
Comment (max-hoffman) [inline sql/parse/parse.go:1299]: is there any way to test this better? the tests added above are mostly just error checks, would be nice to confirm that either the positive case works as expected and/or this doesn't have untested edge cases. Even just unit tests that just make sure the reordering works.

File: sql/parse/parse_test.go [hunk 2]
```diff
@@ -2873,6 +2873,20 @@ CREATE TABLE t2
 				"foo",
 			),
 		},
+		{
+			input: `alter table t add index (i), drop index i, add check (i = 0), drop check chk, drop constraint c, add column i int, modify column i text, drop column i, rename column i to j`,
```
Comment (max-hoffman) [inline sql/parse/parse_test.go:2877]: if i did something to break this test, is there a docstring you could add to `convertMultiAlterDDL` that would help me figure out the precedence order? It seems like the order of top-level case statements are the precedence order? and there's no real mysql docs for this behavior?

**Review by max-hoffman — APPROVED** [review]
LGTM, one more docstring might bring people up to speed quickly with what you learned implementing this
